accounts-db/src/tiered_storage.rs

```diff
@@ -163,6 +165,11 @@ impl TieredStorage {
     pub fn is_empty(&self) -> bool {
         self.len() == 0
     }
+
+    pub fn capacity(&self) -> u64 {
+        self.reader()
+            .map_or(MAX_TIERED_STORAGE_FILE_SIZE, |reader| reader.capacity())
```

Comment:
I think here we need MAX_TIERED_STORAGE_FILE_SIZE.  Because the or case means we don't have a reader yet, meaning we don't know whether it is cold or hot yet.

But yep, under the HotStorageReader it should return MAX_HOT_STORAGE_FILE_SIZE.

Reply:
> I think this is another example of the API being inverted/backwards.

My original thinking is that: the accounts-db layer manages temperature (i.e., hot / cold / warm .. etc) as it has the whole picture of active / inactive accounts and other related information like shrink, and the accounts-file layer's job is to determine the best suitable file format based on the temperature assigned by the accounts-db. (and this is the reason why it was TieredHot previously as we eventually need an enum to determine the temperature.)

So back to the main topic, if we think determining temperature is accounts-db layer's job, then we should have some temperature information here even without constructing a reader.

We can revisit this later as we currently don't have plans for cold format.  For now I think MAX_TIERED_STORAGE_FILE_SIZE is good enough.